vortex-array/src/arrays/struct_/serde.rs

```diff
@@ -53,10 +53,11 @@ impl EncodingVTable for StructEncoding {
             })
             .try_collect()?;
 
-        Ok(
-            StructArray::try_new(struct_dtype.names().clone(), children, len, validity)?
-                .into_array(),
-        )
+        let DType::Struct(struct_dtype, _) = dtype else {
+            unreachable!()
+        };
+
+        Ok(StructArray::try_new_with_dtype(children, struct_dtype, len, validity)?.into_array())
```

Comment:
You could just clone the `struct_dtype` here.

Reply:
I think there are 2 options here. We move this to the begining of the function and replace the existing `as_struct()` call or just call clone here. You will end up doing one extra increment but I thought it's worth it for simplicity

Reply:
Yep, I moved it around (opt. 1)